src/main/java/org/purpurmc/purpurextras/modules/LoomMaxLayersModule.java

```diff
@@ -0,0 +1,128 @@
+import org.bukkit.entity.Player;
+import org.bukkit.event.EventHandler;
+import org.bukkit.event.EventPriority;
+import org.bukkit.event.Listener;
+import org.bukkit.event.inventory.InventoryClickEvent;
+import org.bukkit.inventory.meta.BannerMeta;
+import org.purpurmc.purpurextras.PurpurExtras;
+
+import java.util.HashMap;
+import java.util.List;
+
+/**
+ * If enabled, banners will be able to use more than the limit of 6 layers.
+ * In default config up to 10 banner layers are allowed.
+ */
+public class LoomMaxLayersModule implements PurpurExtrasModule, Listener {
+
+    private final HashMap<Player, List<Pattern>> playerBannerData = new HashMap<>();
+    private int maxLayers;
+
+    protected LoomMaxLayersModule() {
+        this.maxLayers = PurpurExtras.getPurpurConfig().getInt("settings.loom.max-layers", 6);
+        if (this.maxLayers <= 0) this.maxLayers = 6;
+    }
+
+    @Override
+    public void enable() {
+        PurpurExtras plugin = PurpurExtras.getInstance();
+        plugin.getServer().getPluginManager().registerEvents(this, plugin);
+    }
+
+    @Override
+    public boolean shouldEnable() {
+        return this.maxLayers != 6;
+    }
+
+    @EventHandler(priority = EventPriority.NORMAL, ignoreCancelled = true)
+    public void onInventoryClick(InventoryClickEvent event) {
+        Player player = (Player) event.getWhoClicked();
+        ItemStack currentItem = event.getCurrentItem();
+        ItemStack cursorItem = event.getCursor();
+
+        if (!(event.getInventory() instanceof LoomInventory)) return;
+        if (currentItem != null && MaterialSetTag.BANNERS.isTagged(currentItem.getType())) {
+            if (newLayerExceedsMaxLayers(currentItem)) return;
+            switch (event.getSlotType()) {
+                case CRAFTING -> {
+                    if (isPlaceAction(event.getAction())) {
+                        save(player, currentItem);
+                    } else {
+                        undo(player, currentItem);
+                    }
+                }
+                case CONTAINER, QUICKBAR -> {
+                    if (event.getAction() == InventoryAction.MOVE_TO_OTHER_INVENTORY) {
+                        save(player, currentItem);
+                    }
+                }
+                case RESULT -> craftLoad(player, currentItem);
+            }
+        } else if (cursorItem != null && MaterialSetTag.BANNERS.isTagged(cursorItem.getType())) {
+            if (newLayerExceedsMaxLayers(cursorItem)) return;
+            if (event.getSlotType().equals(InventoryType.SlotType.CRAFTING) && isPlaceAction(event.getAction())) {
+                save(player, cursorItem);
+            }
+        }
+    }
+
+    private boolean isPlaceAction(InventoryAction action) {
+        return action == InventoryAction.PLACE_SOME ||
+                action == InventoryAction.PLACE_ALL ||
+                action == InventoryAction.PLACE_ONE;
+    }
+
+    private void save(Player player, ItemStack banner) {
```

Comment:
arbitrary method name. What are we saving?
Don't be afraid of long method names. I wanna know what code I read does from just names